cmd/util/ledger/migrations/storage_v6.go

```diff
@@ -1079,12 +965,43 @@ func (c *ValueConverter) Convert(value oldInter.Value) (result newInter.Value) {
 	c.result = nil
 
 	defer func() {
-		if r := recover(); r != nil {
-			c.migration.missingTypeValues += 1
-			result = nil
+		c.result = prevResult
+
+		r := recover()
+		if r == nil {
+			return
 		}
 
-		c.result = prevResult
+		if typeError, ok := r.(newInter.TypeLoadingError); ok {
+			c.migration.reportFile.WriteString(
+				fmt.Sprintf(
+					"skipped migrating value due to missing static type: owner: %x, type: %s\n",
+					value.GetOwner(),
+					typeError.TypeID,
+				),
+			)
+		} else {
+			err := r.(error)
+			if strings.Contains(err.Error(), "expected `StoragePath`, got `Path`") {
```

Comment:
Maybe we can generalize this, as some errors may also e.g. expect `CapabilityPath`

Reply:
yeah, I'm working on adding more specific logs for different errors, this is also cleaned up in that. Will push the changes soon.

Reply:
Added in 8677eb7